crates/core/src/constraint_system/prove.rs

```diff
@@ -116,8 +116,39 @@ where
 		mut exponents,
 		non_zero_oracle_ids,
 		channel_count,
+		table_size_specs,
 	} = constraint_system.clone();
 
+	if table_sizes.len() != table_size_specs.len() {
```

Comment:
Pls move to separate function if it is possible

Reply:
Sure, will do that in one fell swoop squashing other instances

https://linear.app/irreducible/issue/CRY-484/clean-up-duplication-of-checking-table-sizes